src/ripple/shamap/impl/SHAMapInnerNode.cpp

```diff
@@ -154,39 +156,43 @@ SHAMapInnerNode::makeFullInner(
         ret->hash_ = hash;
     else
         ret->updateHash();
+
     return ret;
 }
 
 std::shared_ptr<SHAMapTreeNode>
 SHAMapInnerNode::makeCompressedInner(Slice data)
 {
-    Serializer s(data.data(), data.size());
+    // A compressed inner node is serialized as a series of 33 byte chunks,
+    // representing a one byte "position" and a 256-bit hash:
+    constexpr std::size_t chunkSize = uint256::bytes + 1;
 
-    int len = s.getLength();
+    if (auto const s = data.size();
+        (s % chunkSize != 0) || (s > chunkSize * branchFactor))
+        Throw<std::runtime_error>("Invalid CI node");
+
+    SerialIter si(data);
 
     auto ret = std::make_shared<SHAMapInnerNode>(0, branchFactor);
 
-    auto retHashes = ret->hashesAndChildren_.getHashes();
-    for (int i = 0; i < (len / 33); ++i)
-    {
-        int pos;
+    auto hashes = ret->hashesAndChildren_.getHashes();
 
-        if (!s.get8(pos, 32 + (i * 33)))
-            Throw<std::runtime_error>("short CI node");
+    while (!si.empty())
+    {
+        auto const hash = si.getBitString<256>();
+        auto const pos = si.get8();
 
         if ((pos < 0) || (pos >= branchFactor))
```

Comment:
gcc warns here:
```
warning: comparison is always false due to limited range of data type [-Wtype-limits]
```
Yes, I know this isn't new code; I'm not sure why it didn't use to warn, but we probably want to fix the warning anyway. (pos is unsigned, so it can't be less than 0).